packages/toolpad-app/src/runtime/PreviewHeader.tsx

```diff
@@ -1,35 +1,48 @@
 import * as React from 'react';
-import { Stack, Button, Typography } from '@mui/material';
+import { Button, Typography, Box, useTheme, Alert } from '@mui/material';
 import EditIcon from '@mui/icons-material/Edit';
-import Header from '../toolpad/ToolpadShell/Header';
-import { ThemeProvider } from '../ThemeContext';
+import { useMatch } from 'react-router-dom';
+import { IS_CUSTOM_SERVER, PREVIEW_HEADER_HEIGHT } from './constants';
 
-export interface PreviewHeaderProps {
-  pageId?: string;
-}
+export default function PreviewHeader() {
+  const pageMatch = useMatch('/pages/:slug');
+  const activePage = pageMatch?.params.slug;
+
+  const theme = useTheme();
 
-export default function PreviewHeader({ pageId }: PreviewHeaderProps) {
   return (
-    <ThemeProvider>
-      <Header
-        enableUserFeedback={false}
-        actions={
-          <Stack direction="row" spacing={1} alignItems="center">
-            <Typography variant="body2" sx={{ color: 'primary.main' }}>
-              This is a preview version of the application.
-            </Typography>
+    <Box
+      sx={{
+        position: 'fixed',
+        width: '100%',
+        height: PREVIEW_HEADER_HEIGHT,
+        zIndex: theme.zIndex.drawer + 1,
+      }}
+    >
+      <Alert
+        severity="warning"
+        // variant="filled"
+        sx={{
+          borderRadius: 0,
+        }}
+        action={
+          IS_CUSTOM_SERVER ? null : (
             <Button
-              variant="outlined"
+              color="inherit"
+              size="small"
               endIcon={<EditIcon />}
-              color="primary"
               component="a"
-              href={pageId ? `/_toolpad/app/pages/${pageId}` : '/_toolpad/app'}
+              href={activePage ? `/_toolpad/app/pages/${activePage}` : '/_toolpad/app'}
             >
               Edit
             </Button>
-          </Stack>
+          )
         }
-      />
-    </ThemeProvider>
+      >
+        <Typography variant="body2">
+          This is a preview version of the application. It&apos;s not suitable for production.
```

Comment:
```suggestion
          This is a preview version of the application, not suitable for production. Click "Edit" to return to the editor.
```
```suggestion
          This is a preview version of the application, not suitable for production. Click "Edit" to go to the editor.
```

Reply:
I'll change the button text to be more actionable instead:

<img width="719" alt="Screenshot 2023-11-02 at 09 18 10" src="https://github.com/mui/mui-toolpad/assets/2109932/51b0cab4-4d0d-4594-9cbc-eb6407d0130f">